src/ScatterChart.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Chart;
+
+// use Atk4\Ui\JsExpression;
+
+class ScatterChart extends Chart
+{
+    public $type = ChartType::TYPE_SCATTER;
+
+    /** @var string X field name */
+    protected $xField;
+
+    /** @var string Y field name */
+    protected $yField;
+
+    /** @var string R (radius) field name */
+    protected $rField;
+
+    public function prepareDatasets(): void
+    {
+        $columns = $this->columns;
+
+        $label = array_shift($columns) ?? null;
+        $this->xField = array_shift($columns) ?? 'x';
+        $this->yField = array_shift($columns) ?? 'y';
+        $this->rField = array_shift($columns) ?? 'r';
+
+        // initialize data-set
```

Comment:
I prefer no comments, this is obvious what it does

Reply:
I'm still thinking about splitting initialization and data fill and moving data fill to later stages (before render). Not sure if that will go into this PR, probably not yet, but that's why I have these comments there to easily see that these blocks "are asking" to be split :)